src/app/home/pages/PlansPage/components/PlansTable.tsx

```diff
@@ -0,0 +1,198 @@
+
+interface IPlansTableProps {
+  planList: any[]; // TODO
+  addPlanDisabledObj: IAddPlanDisabledObjModel;
+  toggleWizardOpen: () => void;
+}
+
+interface IExpandedCells {
+  [planName: string]: number; // Index of expanded column
+}
+
+const PlansTable: React.FunctionComponent<IPlansTableProps> = ({
+  planList,
+  addPlanDisabledObj,
+  toggleWizardOpen,
+}: IPlansTableProps) => {
+  const [expandedCells, setExpandedCells] = useState<IExpandedCells>({});
+
+  const columns = [
+    { title: 'Name', transforms: [sortable] },
+    {
+      title: 'Migrations',
+      transforms: [sortable],
+      cellTransforms: [compoundExpand],
+    },
+    { title: 'Source', transforms: [sortable] },
+    { title: 'Target', transforms: [sortable] },
+    { title: 'Replication repository', transforms: [sortable] },
+    { title: 'PVs', transforms: [sortable] },
+    { title: 'Last state', transforms: [sortable] },
+    '',
+  ];
+
+  // TODO add type for plan?
+  const getSortValues = (plan) => {
+    const {
+      planName,
+      migrationCount,
+      sourceClusterName,
+      targetClusterName,
+      storageName,
+      pvCount,
+      statusText,
+    } = getPlanInfo(plan);
+    return [
+      planName,
+      migrationCount,
+      sourceClusterName,
+      targetClusterName,
+      storageName,
+      pvCount,
+      statusText,
+      '',
+    ];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(planList, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  useEffect(() => setPageNumber(1), [sortBy]);
+
+  const rows = flatten(
+    currentPageItems.map((plan, planIndex) => {
+      const {
+        planName,
+        migrationCount,
+        sourceClusterName,
+        targetClusterName,
+        storageName,
+        pvCount,
+      } = getPlanInfo(plan);
+      return [
+        {
+          meta: { planName },
+          isOpen: Object.keys(expandedCells).includes(planName),
+          cells: [
+            {
+              title: planName,
+              props: { component: 'th' },
+            },
+            {
+              title: (
+                <span className={classNames('pf-c-icon', { 'pf-m-info': migrationCount > 0 })}>
+                  <MigrationIcon /> {migrationCount}
+                </span>
+              ),
+              props: {
+                isOpen: expandedCells[planName] === 1,
+                ariaControls: 'migrations-history-expansion-table',
+              },
+            },
+            sourceClusterName,
+            targetClusterName,
+            storageName,
+            pvCount,
+            {
+              title: <PlanStatus plan={plan} />,
+            },
+            {
+              title: <PlanActions plan={plan} />,
+              props: {
+                className: 'pf-c-table__action',
+              },
+            },
+          ],
+        },
+        {
+          parent: planIndex * 2,
```

Comment:
This line still makes my head hurt :)

Reply:
ha, yeah, I think maybe it's worth at least adding a comment there. I'll think about how we can make that more readable.

Reply:
Added a comment to clarify the math.
```
// Plan 0 => rows 0 and 1, Plan 1 => rows 2 and 3, Plan 2 => rows 4 and 5, etc.
```